packages/caliper-core/lib/worker/rate-control/maxRate.js

```diff
@@ -0,0 +1,203 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+const RateInterface = require('./rateInterface.js');
+const Sleep = require('../../common/utils/caliper-utils').sleep;
+const Logger = require('../../common/utils/caliper-utils').getLogger('maxRate.js');
+
+/**
+ * Rate controller for driving at a maximum TPS.
+ */
+class MaxRate extends RateInterface {
+    /**
+     * Creates a new instance of the MaxRate class.
+     * @constructor
+     * @param {object} opts Options for the rate controller.
+     */
+    constructor(opts) {
+        super(opts);
+
+        // Map for TPS observations
+        this.observedTPS = new Map();
+        // Map for TPS settings
+        this.tpsSettings = new Map();
+        // MPS for observed stats
+        this.statisticsMap = new Map();
+    }
+
+    /**
+     * Initialise the rate controller with a passed msg object
+     * @param {object} msg Client options with adjusted per-client load settings.
+     * @param {string} msg.type The type of the message. Currently always 'test'
+     * @param {string} msg.label The label of the round.
+     * @param {object} msg.rateControl The rate control to use for the round.
+     * @param {number} msg.trim The number/seconds of transactions to trim from the results.
+     * @param {object} msg.args The user supplied arguments for the round.
+     * @param {string} msg.cb The path of the user's callback module.
+     * @param {string} msg.config The path of the network's configuration file.
+     * @param {number} msg.numb The number of transactions to generate during the round.
+     * @param {number} msg.txDuration The length of the round in SECONDS.
+     * @param {number} msg.totalClients The number of clients executing the round.
+     * @param {number} msg.clients The number of clients executing the round.
+     * @param {object} msg.clientArgs Arguments for the client.
+     * @param {number} msg.clientIdx The 0-based index of the current client.
+     * @param {number} msg.roundIdx The 1-based index of the current round.
+     *
+     * @async
+     */
+    async init(msg) {
+        this.sleepTime = 100;
+        this.reset = false;
```

Comment:
What's the purpose of `reset`?

Reply:
oops - that was left over from the prototype .. my bad!